pubsub/rabbitpubsub/rabbit.go

```diff
@@ -592,8 +592,11 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 			}
 
 		case <-maxTime:
-			// Timed out. Return whatever we have.
-			return ms, nil
+			// Timed out. Return whatever we have, or continue waiting if
```

Comment:
It looks to me like this was right before — wait a little bit if there are no messages, then return — and you changed it to wait indefinitely until it gets a message. In other words, the change here is nearly the opposite of the one you made to mempubsub. 

You don't seem to mention this change in your commit description, so I'm not sure what your intent was.

Reply:
I believe the behavior is the same before this change and after it.

Before this change, it would return and `pubsub.go` would just call it right back and it will re-enter the `for` loop.

After this change, it will just re-enter the `for` loop directly.

This seems simpler to me, but I don't feel strongly about it, so I'll put it back if you prefer.

It's different than `mempubsub` because it already has a `for` loop, so it's easy to continue looping; by returning, `mempubsub` doesn't need the `for` loop at all.